### DISCUSSION TITLE
Add ambient occlusion to increase details for eg. bas relief

### DISCUSSION THREAD

**Discussion by umutucler** [thygate/stable-diffusion-webui-depthmap-script#327] [Ideas]
Any possibility to add ambient occlusion to increase details for creating bas relief?

Creating a bas relief with depth map and a multiplied Ambient Occlusion (AO) will add gradient shadows which increases the level of details in the 3d model.

### Image examples

**Depthmap**
![Depthmap](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/5963503/2c8489ef-44db-4535-974f-ff56431eb0d5)

**AO**
![Ambient_Occlusion](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/5963503/36013c5c-4879-496e-929b-dc0a655601a9)

**Depthmap only vs depthmap with multiplied AO (levels are modified to increase black and white)**
![AO+Depth_multiplied](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/5963503/83e7836f-f2fd-4d9c-8e60-32f020f48a2d)

### 3D results

**Depthmap**
![Render_depth](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/5963503/c425ac76-a9cf-472a-ad8c-842be4410217)

**Depthmap with multiplied AO (levels are modified to increase black and white)**
![Render_Depth+AO](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/5963503/b051e892-a597-45a9-99f4-5cf9bacab247)

As you can see, adding AO to depth maps increases level of detail and can be tweaked in 3D software to look even more realistic.

I found this paper https://www.theorangeduck.com/media/uploads/other_stuff/nnao.pdf which could be inspiration to Screen Space Ambient Occlusion (SSAO)

Here is another algorithm https://www.theorangeduck.com/page/pure-depth-ssao but I'm not sure if it can be used for 2D images

**Comment by graemeniedermayer** [comment]
Usually ambient occlusion is something you'd add after with tools like blender. If you have some pictures that might help demonstrate what you mean. Also I might be misunderstand your use-case/workflow.

For intricate depth maps one things I've found useful is using multiple resolution depth maps and then mixing them together.

**Reply by umutucler** [reply]
Thanks @graemeniedermayer, I have updated the post with images and details.

Regarding adding AO in Blender, as far as I know, you can only do this for 3D geometry. Blender can't add AO to 2D images or depthmaps for that matter. I haven't tried using multiple resolutions for intricate depth maps but will try it out. Can you give me an example of resolution to try with? Min + max?

**Reply by graemeniedermayer** [reply]
Yes so you can obtain somewhat similar results with resolution mixing. So this is changing the algorithm resolution (the output image will stay the same resolution). This example was done with zoedepth. The new standalone mode can also help if you run into vram issues

![differentResolutions](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/38518120/772ed786-b4da-40ed-ba55-9974c7058a3b)

Opacity Mixing

![mixedResults](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/38518120/cb1391d2-6321-4016-894a-29757f680268)

This process is sort of what boost does automatically although boost has memory/time trouble with the bigger models (midas512/zoedepth).

I definitely could see how 2d AO could be useful especially for controlnet or depthmasking

**Reply by graemeniedermayer** [reply]
Also super thanks for this paper https://www.theorangeduck.com/media/uploads/other_stuff/nnao.pdf . I've been using AI normal maps and I haven't been super happy with edges which this seems to have some interesting clues.

**Reply by umutucler** [reply]
I tried resolution mixing with a portrait photo of a person but didn't really get a better result. 

I found another paper you might find interesting. Here is an example of depth map:
![Depth-map-processing-An-example-of-depth-map-interpolation-for-RGB-D-scans-based-on-the](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/5963503/94e4968b-edbf-4a0a-9cbb-33bcdad06953)

The paper in PDF:
[Facial_image_super-resolution_guided_by_adaptive_g.pdf](https://github.com/thygate/stable-diffusion-webui-depthmap-script/files/12497500/Facial_image_super-resolution_guided_by_adaptive_g.pdf)
